rfcs/text/????-hook-service.md

```diff
@@ -0,0 +1,134 @@
+
+It was also used by the embedding application (HASH) to know that any particular block had an editable text section within it. The particular solution HASH uses for rich text editing needs (or needed) to know at the time blocks are loaded that a user can edit text within them. This worked because a block would use `editableRef` by reading it from the properties passed to it, which means the `editableRef` key would be in the block's schema.
+
+`editableRef` is an unsuitable candidate for standardisation for a number of reasons:
+- It does not indicate which block entity property on which to store the rich text data
+- It means that the type of the value in the `text` field is unknown to the block, and may not match the value in the block's schema
+- It has a one time use – you cannot use it to make multiple editable islands within a block (think a blockquote with a caption)
+- The requirement for the use of the feature to show up in the block's schema can make certain kinds of blocks impossible (i.e, blocks that may or may not have editable sections depending on their current state).
+- An embedding application can only provide one implementation for `editableRef`, which means there may be a mismatch between the implementation provided by the embedding application and the feature expected by the block. I.e, an embedding application may use `editableRef` to provide image editing features, whereas the block may expect it to be used to provide rich text editing features. 
+
+Another minor reason is its naming is tied into React – i.e, "editableRef". 
+
+This proposal fixes all of these problems.
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+- Is hook the best name for this service? Other options include (not exhaustive):
+  - editable/editing/editor service
+  - view service
+  - UI service
+  - plugin service
+  - DOM service
+  - extension service
+- Is the name 'hook' too tied into React and therefore confusing? 
+- Is there any functionality expected to be enabled by this service where an embedding application would need to know statically that the service is being used – i.e, does this need to show up in the schema. If so, how does this work with dynamic properties (i.e, a property on a value in a list)
+- Is the use of property paths to indicate the relevant value problematic? Do we need to make it easy to generate these property paths (possible using a proxy)?
+- Will blocks want to pass up data which is not present on `blockEntity` (i.e, on a linked entity or aggregation) or not provided by the block service at all
+- Do we want to standardise the "types" of hook available or do we want it to be any arbitrary string? Should we 'special-case' the recommended types by providing specific message types for those, or should all uses of the service use the same message type? 
+- Is the recommended list of types of hook sufficient?
```

Comment:
To be clear, these are the initial types we expect will be commonly used.

Once we have witnessed real-world usage of the hook types, we will update this section to reflect reality.

Reply:
Do you think I need to add this to the RFC text?

Reply:
No. What I mean to imply with my comment above, though, is that I don't think it needs to be a question/something we spend further time discussing now. The line can be removed.

Reply:
Ah, gotcha. Will do!